src/ResourceManager/Reservations/Commands.Reservations/Cmdlets/MergeReservation.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Azure.Commands.Reservations.Common;
+using Microsoft.Azure.Management.Reservations.Models;
+using System.Management.Automation;
+using Microsoft.Azure.Management.Reservations;
+using Microsoft.Azure.Commands.Reservations.Models;
+
+namespace Microsoft.Azure.Commands.Reservations.Cmdlets
+{
+    [Cmdlet(VerbsData.Merge, "AzureRmReservation", DefaultParameterSetName = Constants.ParameterSetNames.CommandParameterSet, SupportsShouldProcess = true), OutputType(typeof(List<PSReservation>))]
+    public class MergeReservation : AzureReservationsCmdletBase
+    {
+        [Parameter(ParameterSetName = Constants.ParameterSetNames.CommandParameterSet, 
+            Mandatory = true)]
+        [ValidateNotNull]
+        public string ReservationOrderId { get; set; }
+
+        [Parameter(ParameterSetName = Constants.ParameterSetNames.CommandParameterSet, 
+            Mandatory = true)]
+        [ValidateNotNull]
+        [ValidateCount (2, 2)]
+        public string[] ReservationId { get; set; }
+
+        [Parameter(ParameterSetName = Constants.ParameterSetNames.ObjectParameterSet,
+            Mandatory = true,
+            ValueFromPipeline = true)]
```

Comment:
How can this come from the pipeline?

Reply:
usually user would be using command parameters but we added this for possible pipeline scenario

Reply:
You should remove this property - users will need to provide two reservations, in order, o that this value cannot come from the pipeline